rand/uuid.go

```diff
@@ -5,6 +5,23 @@ import (
 	"io"
 )
 
+// UUIDIdempotencyToken provides a utility to get idempotency tokens in the
+// UUID format.
+type UUIDIdempotencyToken struct {
+	uuid *UUID
+}
+
+// NewUUIDIdempotencyToken returns a idempotency token provider returning
+// tokens in the UUID random format using the reader provided.
+func NewUUIDIdempotencyToken(r io.Reader) *UUIDIdempotencyToken {
+	return &UUIDIdempotencyToken{uuid: NewUUID(r)}
+}
+
+// GetIdempotencyToken returns a random UUID value for Idempotency token.
+func (u UUIDIdempotencyToken) GetIdempotencyToken() (string, error) {
+	return u.uuid.GetUUID()
+}
+
```

Comment:
Shouldn't this be used when in `func(o *Options) GetIdempotencyToken() (string, error)`, don't see it being used. Can you verify it's generated?

Reply:
Ah so the generated helper doesn't need to use a  `GetIdempotencyToken` interface because the helper is defined in the same package as the Client. This allows the helper function to make use of the `Options` type directly instead of being forced to use an interface.  https://github.com/aws/aws-sdk-go-v2/pull/628/files?file-filters%5B%5D=.go#diff-9ced4025b14d09f3f49aafe35890b89eR91-R94

This is the reason there is no `GetIdempotencyToken()` call.

Reply:
Gotcha. I think NewUUIDIdempotencyToken will be used as default value when we have default client config options.